cfme/tests/services/test_rest_services.py

```diff
@@ -116,121 +137,156 @@ def test_edit_multiple_services(self, rest_api, services):
+        """Test retiring a service now.
 
-    def test_retire_service_now(self, rest_api, services):
-        """Test retiring a service
-        Prerequisities:
-            * An appliance with ``/api`` available.
-        Steps:
-            * Retrieve list of entities using GET /api/services , pick the first one
-            * POST /api/service/<id> (method ``retire``)
         Metadata:
             test_flag: rest
         """
-        assert "retire" in rest_api.collections.services.action.all
-        retire_service = services[0]
-        retire_service.action.retire()
-        wait_for(
-            lambda: not rest_api.collections.services.find_by(name=retire_service.name),
-            num_sec=600,
-            delay=10,
-        )
+        if from_detail:
+            for service in services:
+                service.action.retire()
+                assert rest_api.response.status_code == 200
+        else:
+            rest_api.collections.services.action.retire(*services)
+            assert rest_api.response.status_code == 200
+
+        for service in services:
+            wait_for(
+                lambda: not rest_api.collections.services.find_by(name=service.name),
+                num_sec=600,
+                delay=10,
+            )
+
+    @pytest.mark.parametrize(
+        "from_detail", [True, False],
+        ids=["from_detail", "from_collection"])
+    def test_retire_service_future(self, rest_api, services, from_detail):
+        """Test retiring a service in future.
 
-    def test_retire_service_future(self, rest_api, services):
-        """Test retiring a service
-        Prerequisities:
-            * An appliance with ``/api`` available.
-        Steps:
-            * Retrieve list of entities using GET /api/services , pick the first one
-            * POST /api/service/<id> (method ``retire``) with the ``retire_date``
         Metadata:
             test_flag: rest
         """
-        assert "retire" in rest_api.collections.services.action.all
-
-        retire_service = services[0]
-        date = (datetime.datetime.now() + datetime.timedelta(days=5)).strftime('%m/%d/%y')
+        date = (datetime.datetime.now() + datetime.timedelta(days=5)).strftime("%Y/%m/%d")
         future = {
             "date": date,
             "warn": "4",
         }
-        date_before = retire_service.updated_at
-        retire_service.action.retire(future)
 
-        def _finished():
-            retire_service.reload()
-            if retire_service.updated_at > date_before:
-                return True
-            return False
+        if from_detail:
+            for service in services:
+                service.action.retire(**future)
+                assert rest_api.response.status_code == 200
+        else:
+            rest_api.collections.services.action.retire(*services, **future)
+            assert rest_api.response.status_code == 200
+
+        def _finished(ser):
```

Comment:
**Optional**
Two things:

* use `service` here instead of `ser` (looks like you made this change other places)
* combine the conditionals for returning False with `or`